Geometry/HGCalCommonData/src/HGCalDDDConstants.cc

```diff
@@ -109,6 +105,12 @@ std::pair<int,int> HGCalDDDConstants::assignCellHexagon(float x,
   double xx(x), yy(y);
   //First the wafer
   int wafer = cellHex(xx, yy, rmax_, hgpar_->waferPosX_, hgpar_->waferPosY_);
+  if (wafer < 0 || wafer >= (int)(hgpar_->waferTypeT_.size())) 
+    edm::LogWarning("HGCalGeom") << "Wafer " << wafer << ":" 
```

Comment:
maybe this warning message could be more instructive instead of just saying error?